vcf/methods.go

```diff
@@ -1,21 +1,66 @@
 package vcf
 
 import (
-	"fmt"
 	"io"
+	"strconv"
 	"strings"
 
 	"github.com/vertgenlab/gonomics/dna"
 )
 
 // String implements the fmt.Stringer interface for easy printing of Vcf with the fmt package.
 func (v Vcf) String() string {
-	return fmt.Sprintf("%s\t%v\t%s\t%s\t%s\t%v\t%s\t%s\t%s\t%s\n", v.Chr, v.Pos, v.Id, v.Ref, strings.Join(v.Alt, ","), v.Qual, v.Filter, v.Info, v.Format, SamplesToString(v.Samples))
+	var buf strings.Builder
+	buf.Grow(256) // Pre-allocate some space for efficiency.
+
+	buf.WriteString(v.Chr)
+	buf.WriteByte('\t')
+	buf.WriteString(strconv.Itoa(v.Pos))
+	buf.WriteByte('\t')
+	buf.WriteString(v.Id)
+	buf.WriteByte('\t')
+	buf.WriteString(v.Ref)
+	buf.WriteByte('\t')
+	buf.WriteString(strings.Join(v.Alt, ","))
+	buf.WriteByte('\t')
+	buf.WriteString(strconv.FormatFloat(v.Qual, 'f', -1, 64))
+	buf.WriteByte('\t')
+	buf.WriteString(v.Filter)
+	buf.WriteByte('\t')
+	buf.WriteString(v.Info)
+
+	if len(v.Format) > 0 {
+		buf.WriteByte('\t')
+		buf.WriteString(strings.Join(v.Format, ":"))
+		buf.WriteByte('\t')
+		buf.WriteString(SamplesToString(v.Samples))
+	}
+	buf.WriteByte('\n')
+	return buf.String()
 }
 
 // String implements the fmt.Stringer interface for easy printing of Sample with the fmt package.
 func (s Sample) String() string {
-	return sampleToString(s)
+	var buf strings.Builder
+	if s.FormatData == nil {
+		buf.WriteByte('.')
+		return buf.String()
+	}
+	if s.Alleles == nil {
+		buf.WriteByte('.')
+	} else {
+		for i := 0; i < len(s.Alleles); i++ {
+			if i > 0 && i < len(s.Phase) {
```

Comment:
I am interpreting this code as pulling the middle element in the Alleles slice and always interpreting that as the a phase element. If that is correct, there may be a bug here. VCF specs support polyploid input so the genotype field can be something like `1/1|2/3`. In these cases I believe this code would panic.

Reply:
Might be good to add a test case for that as well. Maybe also a test case for when the allele number is double digits. I am pretty sure double digit alleles are handled correctly when you are building the Alleles slice, but I'd feel better if we had a test for it.

Reply:
Could you give me a couple of string representation of the scenarios you are talking about? Not totally sure what I would see
- when you say genotype field can be: `./.` or just `.`
- allele number is double digits. (do you mean it's possible to see 10/11?)

Reply:
Added TODO(s) for the cases `.` and `./.`